openmc/material.py

```diff
@@ -311,6 +327,65 @@ def remove_nuclide(self, nuclide):
         if nuclide._name in self._nuclides:
             del self._nuclides[nuclide._name]
 
+    def add_macroscopic(self, macroscopic):
+        """Add a macroscopic to the material
+
+        Parameters
+        ----------
+        macroscopic : str or openmc.macroscopic.Macroscopic
+            Macroscopic to add
+
+        """
+
+        # Ensure no nuclides, elements, or sab are added since these would be
+        # incompatible with macroscopics
+        if ((len(self._nuclides.keys()) != 0) and
+            (len(self._elements.keys()) != 0) and (len(self._sab) != 0)):
```

Comment:
I would recommend shortening this to:

    if self._nuclides or self._elements or self._sab: